### PR TITLE
Fix definition for optionRenderer in react-select

### PR DESCRIPTION
**Pull request by rcchen**
Source code reference: https://github.com/JedWatson/react-select/blob/18758f78655b4004a3a22fc049f5706cd25bffcd/examples/src/components/CustomRender.js#L19

### CODE REVIEW

**Comment by rcchen** [comment]
@giladgray mentioned there are a bunch of other callbacks that have incorrect definitions, so I'm going to go ahead and batch them all into this PR, so hold off on this one for a bit

File: react-select/react-select-tests.tsx [hunk 1]
```diff
@@ -14,8 +14,11 @@ class SelectTest extends React.Component<React.Props<{}>, {}> {
         const options: ReactSelect.Option[] = [{ label: "Foo", value: "bar" }]
         const onOpen = () => { return; };
         const onClose = () => { return; };
+        const optionRenderer = (option: ReactSelect.Option) => {
+            return <span>{option.label}</span>
```
Comment (giladgray) [inline react-select/react-select-tests.tsx]: could be inlined? `(option: ReactSelect.Option) => <span>{option.label}</span>`

**Comment by giladgray** [comment]
:+1: LGTM!

**Comment by vvakame** [comment]
@rccben please rebase master/HEAD and `push -f`. this pr constains so many file changes...

**Comment by rcchen** [comment]
@vvakame

**Pull request merged**